api/src/database/db.ts

```diff
@@ -314,12 +314,21 @@ export const getDBConnection = function (keycloakToken: object): IDBConnection {
    */
   const _setUserContext = async () => {
     const userGuid = getUserGuid(_token);
+    const userIdentifier = getUserIdentifier(_token);
     const userIdentitySource = getUserIdentitySource(_token);
+    defaultLog.debug({ label: '_setUserContext', userGuid, userIdentifier, userIdentitySource });
 
-    if (!userGuid || !userIdentitySource) {
+    if (!userGuid || !userIdentifier || !userIdentitySource) {
       throw new ApiGeneralError('Failed to identify authenticated user');
     }
 
+    // Patch user GUID
+    const patchUserGuidSqlStatement = queries.database.patchUserGuidSQL(userGuid, userIdentifier, userIdentitySource);
+
+    if (!patchUserGuidSqlStatement) {
```

Comment:
Is this check needed? On line 321 the variable are checked before used in the queries.database.patchUserGuidSQL where they are checked again. Then you are covering a case that will not exists with this check.